modulegen/main.go

```diff
@@ -375,3 +345,12 @@ func runGoCommand(cmdDir string, args ...string) error {
 	cmd.Dir = cmdDir
 	return cmd.Run()
 }
+
+func getRootDir() (string, error) {
```

Comment:
I think we can have a `NewContext(dir string) (Context, error)` function in the `context.go` file, instead of this one. In the end, we are always using it to get the dir, assert there is no error and create a context.

Reply:
Do you want this in this PR ?
I can add this in the new one as it will be a small one.

Reply:
I think it's fine to have this change in this PR, as it's very narrowed to the PR itself. So I'd prefer having it here if possible 🙏

Reply:
I have created NewContext() but for the moment I cannot remove getRootDir as getModulesOrExamples still relies on it.
It shall be easier once getModulesOrExamples is replaced by ctx.GetModules()

Reply:
@mdelapenya ,
Can you please trigger a retry on the workflow ?